UHResidentInformationAPI.Tests/V1/Gateways/UHGatewayTests.cs

```diff
@@ -1,51 +1,128 @@
+
+            var response = _classUnderTest.GetResidentById(houseRef, personNo);
+
+            response.Should().BeNull();
+        }
 
-        // [Test]
-        // public void GetEntityByIdReturnsEmptyArray()
-        // {
-        //     var response = _classUnderTest.GetEntityById(123);
+        [Test]
+        public void GetResidentByIdReturnsPersonalDetails()
+        {
+            var databasePersonEntity = AddPersonRecordToDatabase();
+            var response = _classUnderTest.GetResidentById(databasePersonEntity.HouseRef, databasePersonEntity.PersonNo);
+
+            response.HouseReference.Should().Be(databasePersonEntity.HouseRef);
+            response.FirstName.Should().Be(databasePersonEntity.FirstName);
+            response.LastName.Should().Be(databasePersonEntity.LastName);
+            response.NINumber.Should().Be(databasePersonEntity.NINumber);
+            response.DateOfBirth.Should().Be(databasePersonEntity.DateOfBirth.ToString("O", CultureInfo.InvariantCulture));
+            response.Should().NotBe(null);
+        }
 
-        //     response.Should().BeNull();
-        // }
+        [Test]
+        public void GetResidentByIdReturnsAddressDetails()
+        {
+            var databasePersonEntity = AddPersonRecordToDatabase();
+            var databaseAddressEntity = TestHelper.CreateDatabaseAddressForPerson(databasePersonEntity.HouseRef);
 
-        // [Test]
-        // public void GetEntityByIdReturnsCorrectResponse()
-        // {
-        //     var entity = _fixture.Create<Entity>();
-        //     var databaseEntity = DatabaseEntityHelper.CreateDatabaseEntityFrom(entity);
+            UHContext.Addresses.Add(databaseAddressEntity);
+            UHContext.SaveChanges();
 
-        //     DatabaseContext.DatabaseEntities.Add(databaseEntity);
-        //     DatabaseContext.SaveChanges();
+            var expectedDomainAddress = new DomainAddress
+            {
+                PropertyRef = databaseAddressEntity.PropertyRef,
+                AddressLine1 = databaseAddressEntity.AddressLine1,
+                PostCode = databaseAddressEntity.PostCode
+            };
 
-        //     var response = _classUnderTest.GetEntityById(databaseEntity.Id);
+            var response = _classUnderTest.GetResidentById(databasePersonEntity.HouseRef, databasePersonEntity.PersonNo);
 
-        //     databaseEntity.Id.Should().Be(response.Id);
-        //     databaseEntity.CreatedAt.Should().BeSameDateAs(response.CreatedAt);
-        // }
+            response.HouseReference.Should().Be(databasePersonEntity.HouseRef);
+            response.Address.Should().BeEquivalentTo(expectedDomainAddress);
+        }
+
+        [Test]
+        public void GetResidentByIdReturnsPhoneContactDetails()
+        {
+            var databasePersonEntity = AddPersonRecordToDatabase();
+            var databasePhoneEntity = TestHelper.CreateDatabaseTelephoneNumberForPersonId(databasePersonEntity.ContactSid);
```

Comment:
Need to check if this is what we use to link the person table and telephone contact tables.